### PR TITLE
CB-10367 Add meaningful names to heading anchors in Docs & show a link on hover

### PR DESCRIPTION
**Pull request by nikhilkh**
image

### CODE REVIEW

**Comment by nikhilkh** [comment]
@dblotsky to help review

File: www/static/js/docs.js [hunk 1]
```diff
@@ -48,7 +57,10 @@ $(document).ready(function () {
             // add a special class to the anchor for this toc entry
             var anchorName = getAnchorName(i, heading, prefix);
             $('#' + anchorName).addClass('fragment-anchor');
-
+            if (i > 0) { 
+                // the first link is the header of the page - exclude it
```
Comment (dblotsky) [inline www/static/js/docs.js]: Nitpick: please put the comment above the `if`.

File: _config.yml [hunk 2]
```diff
@@ -73,7 +73,9 @@ redcarpet:
         - lax_spacing
 
         - tables
-
+        
+        # enables anhor links
+        - with_toc_data
```
Comment (dblotsky) [inline _config.yml]: Nitpick: please add a space under the option.

File: www/static/js/docs.js [hunk 3]
```diff
@@ -48,7 +57,10 @@ $(document).ready(function () {
             // add a special class to the anchor for this toc entry
             var anchorName = getAnchorName(i, heading, prefix);
             $('#' + anchorName).addClass('fragment-anchor');
-
+            if (i > 0) {
```
Comment (dblotsky) [inline www/static/js/docs.js]: Should we also apply this condition to the above adding of `'fragment-anchor'`?

**Comment by dblotsky** [comment]
Just a suggestion: maybe add some CSS to make the link vanish unless the title is hovered, and add a bit of spacing.

**Comment by nikhilkh** [comment]
Thanks, Dmitry! I fixed the points that you brought up.

**Pull request closed without merging**